src/stdlib_child_list.f90

```diff
@@ -0,0 +1,369 @@
+        
+        length = this_child_list%num_nodes
+    
+    end function get_length
+
+    
+    !> Changes the size of the input child list to 'length'
+    pure subroutine set_length ( this_child_list, length )
+        class(child_list), intent(inout)  :: this_child_list
+        integer, intent(in)      :: length
+    
+        this_child_list%num_nodes = length
+    
+    end subroutine set_length
+
+    
+    
+    !> Replaces the item stored in node at 'node_index' of the input child list with 'new_item'
+    pure subroutine replace_at_index( this_child_list, item ,node_index )
+
+        class(child_list), intent(inout) :: this_child_list
+        integer, intent(in)        :: node_index
+        class(*), intent(in)       :: item
+        type(node), pointer        :: current_node
+        integer :: index
+
+
+        ! This index will be reference for child list
+        index = node_index;
+
+        ! return if the given node index is not in range of 1 to size of child list
+        if(index<1 .or. index>this_child_list%num_nodes) return;
+
+
+        ! Iterating through parent nodes while size of the child list is smaller than index
+        current_node => this_child_list%head;
+        do while(index>1)
+            index = index-1;
+            current_node => current_node%next;
+        end do
+        current_node%item = item
+
+    end subroutine replace_at_index
+
+    !> Reverses the input child list
+    pure subroutine reverse_child_list (this_child_list)
+        class(child_list), intent(inout) :: this_child_list
+        type(node), pointer        :: temp_node
+        type(node), pointer        :: curr_node
+
+        nullify(temp_node)
+
+        ! Swapping head of the child node with tail of the child node
+        curr_node => this_child_list%head
+        do while (associated(curr_node))
+            temp_node => curr_node%prev;
+            curr_node%prev => curr_node%next;
+            curr_node%next => temp_node;
+            curr_node => curr_node%prev;
+        end do
+
+        temp_node=> this_child_list%head
+        this_child_list%head => this_child_list%tail
+        this_child_list%tail => temp_node
+
+    end subroutine reverse_child_list
+
+    !> Destroy the whole given linked list
+    !> Free  the allocated memory
+    !> Nullify all the variables 
+    pure subroutine destroy_whole_child_list( this_child_list )
+        !Entrada:
+        class(child_list), intent(inout) :: this_child_list
+        !Local:
+        type(node), pointer:: current_node
+
+        do while (this_child_list%num_nodes>0)
+            current_node => this_child_list%head
+            if (associated(current_node%next)) then
+            nullify(current_node%next%prev)
+            this_child_list%head => current_node%next
+            end if
+            call current_node%clear()
+            deallocate(current_node)
+            this_child_list%num_nodes = this_child_list%num_nodes - 1
+        end do
+
```

Comment:
```suggestion
nullify (this_list%head, this_list%tail)
```

It seems that the head and tail of the linked list need to be blanked here, otherwise an error will be reported when a new push node is pushed again.